source/backend/api/Services/DispositionFileService.cs

```diff
@@ -246,6 +246,34 @@ public PimsDispositionSale GetDispositionFileSale(long dispositionFileId)
             return _dispositionFileRepository.GetDispositionFileSale(dispositionFileId);
         }
 
+        public PimsDispositionSale UpdateDispositionFileSale(long dispositionFileId, long saleId, PimsDispositionSale dispositionSale)
+        {
+            _logger.LogInformation("Updating disposition file Sale with DispositionFileId: {id}", dispositionFileId);
+            _user.ThrowIfNotAuthorized(Permissions.DispositionEdit);
+
+            var updatedSale = _dispositionFileRepository.UpdateDispositionFileSale(dispositionSale);
+            _dispositionFileRepository.CommitTransaction();
+
+            return updatedSale;
+        }
+
+        public PimsDispositionSale AddDispositionFileSale(long dispositionFileId, PimsDispositionSale dispositionSale)
```

Comment:
does the dispositionsale have an invalid dispositionfileId? otherwise the file id does not need to be passed separately into the service

Reply:
Updated